modules/programs/micro.nix

```diff
@@ -31,13 +38,72 @@ in {
           for supported values.
         '';
       };
+
+      bindings = mkOption {
+        type = jsonFormat.type;
+        default = { };
+        example = literalExpression ''
+          {
+            "Alt-d" = "SpawnMultiCursor";
+            "Escape" = "RemoveAllMultiCursors";
+            "CtrlDown" = "None";
+            "CtrlUp" = "None";
+            "Shift-PageDown" = "SelectPageDown";
+            "Shift-PageUp" = "SelectPageUp";
+          }
+        '';
+        description = ''
+          Configuration written to
+          <filename>$XDG_CONFIG_HOME/micro/bindings.json</filename>. See
+          <link xlink:href="https://github.com/zyedidia/micro/blob/master/runtime/help/keybindings.md"/>
+          for supported values.
+        '';
+      };
+
+      ensurePlugins = mkOption {
+        type = types.listOf types.str;
+        default = [ ];
+        example = literalExpression ''
+          [
+            "aspell"
+          ]
+        '';
+        description = ''
+          Install micro plugins during activation. See
+          <link xlink:href="https://micro-editor.github.io/plugins.html"/>
+          for a listing of available plugins.
+        '';
+
+      };
     };
   };
 
   config = mkIf cfg.enable {
     home.packages = [ pkgs.micro ];
 
+    home.sessionVariables = mkIf cfg.trueColor { MICRO_TRUECOLOR = "1"; };
+
     xdg.configFile."micro/settings.json".source =
       jsonFormat.generate "micro-settings" cfg.settings;
+
+    xdg.configFile."micro/bindings.json".source =
+      jsonFormat.generate "micro-bindings" cfg.bindings;
+
+    home.activation = let
+      mkInstall = pluginName: ''
+        if ! test -d ${config.xdg.configHome}/micro/plug/${
+          lib.escapeShellArg pluginName
+        }; then
+          (set -x
+            $DRY_RUN_CMD ${pkgs.micro}/bin/micro -plugin install ${
+              lib.escapeShellArg pluginName
+            }
+          )
+        fi
+      '';
```

Comment:
Hmm, this doesn't seem very Nixy to me. Is it not possible to package the plugins as Nix packages and, for example, symlink the plugins into place?

This also seems like it would require network access? Typically, the activation script should be possible to run without network availability.

Reply:
I agree. This was a temporary approach. Should i package all the plugins in nixpkgs, or here?

Reply:
We're very much against putting packages in the HM repository 🙂

The ideal is to have them in Nixpkgs. The second best is to have them in some other publicly accessible package repository, which is handy if the package set is automatically generated. See, e.g., the [Firefox add-on package set](https://gitlab.com/rycee/nur-expressions/-/blob/master/pkgs/firefox-addons/generated-firefox-addons.nix).